commit 57c1d5de7d1527c04b5e58089260064b63306b35
Author: Deborah Brouwer <deborahbrouwer3563@gmail.com>
Date:   Wed Dec 8 01:40:42 2021 +0100

    media: vivid: fix timestamp and sequence wrapping
    
    The error injection controls that test wrap-around sequence and timestamp
    counters were partially broken. Add a menu option for 64 or 32 bit signed
    timestamp wrapping. Prevent the timestamp from wrapping around before the
    device can be tested.  Remove the sequence count from the timestamp
    calculation so that sequence wrapping does not interfere with the
    timestamp.  Add consistent time and sequence wrapping to sdr and touch
    devices.
    
    Signed-off-by: Deborah Brouwer <deborahbrouwer3563@gmail.com>
    Signed-off-by: Hans Verkuil <hverkuil-cisco@xs4all.nl>
    Signed-off-by: Mauro Carvalho Chehab <mchehab@kernel.org>

diff --git a/drivers/media/test-drivers/vivid/vivid-touch-cap.c b/drivers/media/test-drivers/vivid/vivid-touch-cap.c
index ebb00b128030..64e3e4cb30c2 100644
--- a/drivers/media/test-drivers/vivid/vivid-touch-cap.c
+++ b/drivers/media/test-drivers/vivid/vivid-touch-cap.c
@@ -262,7 +262,7 @@ void vivid_fillbuff_tch(struct vivid_dev *dev, struct vivid_buffer *buf)
 
 	__s16 *tch_buf = vb2_plane_vaddr(&buf->vb.vb2_buf, 0);
 
-	buf->vb.sequence = dev->touch_cap_seq_count;
+	buf->vb.sequence = dev->touch_cap_with_seq_wrap_count;
 	test_pattern = (buf->vb.sequence / TCH_SEQ_COUNT) % TEST_CASE_MAX;
 	test_pat_idx = buf->vb.sequence % TCH_SEQ_COUNT;